vpr/src/route/rr_graph2.cpp

```diff
@@ -1305,8 +1305,8 @@ bool verify_rr_node_indices(const DeviceGrid& grid,
         auto& rr_node = rr_nodes[size_t(inode)];
 
         if (rr_graph.node_type(inode) == SOURCE || rr_graph.node_type(inode) == SINK) {
-            int rr_width = (rr_node.xhigh() - rr_node.xlow() + 1);
-            int rr_height = (rr_node.yhigh() - rr_node.ylow() + 1);
+            int rr_width = (rr_graph.node_xhigh(rr_node.id()) - rr_graph.node_xlow(rr_node.id()) + 1);
```

Comment:
Consider to use the newly added API node_length()

Reply:
I am also unsure if this instance works for `node_length()`. The `rr_width` and `rr_height` each are only equivalent to `node_length()` in certain cases. (eg. `rr_width == node_length` when the node type is `CHANX`

Reply:
Indeed, the ``node_length()`` should be only applicable to ``CHANX`` and ``CHANY``. 
I double checked the coordinates of ``SOURCE`` and ``SINK`` nodes. They do have a different xlow/ylow/xhigh/yhigh for each node.

https://github.com/verilog-to-routing/vtr-verilog-to-routing/blob/eea0df05c079767cb30f41b5160de17ffedf37dd/vpr/src/route/rr_graph.cpp#L1430-L1432

Therefore, we may consider an API ``node_bounding_box_area()``. But I am not sure such API is frequently needed by router or placer. We can ask @vaughnbetz for advice.

If not frequently used, we can keep the current code changes.